.travis.yml

```diff
@@ -7,6 +7,11 @@ cache:
     # cache folder with Cypress binary
     - ~/.cache
 dist: xenial
+env:
+  global:
+    # Test keys, used by cypress tests
+    - STRIPE_PUBLISHABLE_KEY=pk_test_8z7r9ZpQQcaduYfZlREiNfhP
+    - STRIPE_SECRET_KEY=sk_test_Jtj2Ibwg2DQzT3ZlxF196Dpp
```

Comment:
Instead of adding these keys to the code you can add these keys to travis settings and you can make the keys not visible. 
![screen shot 2019-02-27 at 22 31 39](https://user-images.githubusercontent.com/2124818/53524366-80879d00-3adf-11e9-94f0-00c502b8b783.png)

Reply:
If they are added in the Travis settings, that means they are not passed to pull requests from forks. So, that'd mean we couldn't run the browser tests for people who submit PRs from their fork (me included!).

The keys aren't secret, so by adding the verbatim in the settings, it'll run the browser tests for PRs too.

Reply:
I'll add a comment along these lines to explain why they're there...

Reply:
I did not know that. That seems kind of stupid to be honest... if an environment variable is set it feels like you would want that to be set for all pull requests...

Reply:
Done!

Reply:
The idea is that if you've got an open source project, it lets you set environment variables that are actually secret. If they're run on any pull request, a contributor could craft a pull request that extracts the environment variable by POSTing it to a URL or something.

By only allowing those environment variables to be available on local branches and master, it'll always run on trusted code, therefore it's safe to use actual secrets.

Reply:
I stand corrected! (I wonder if there ever was a security researcher who came up with that).